sympy/functions/combinatorial/tests/test_comb_factorials.py

```diff
@@ -309,11 +309,12 @@ def test_binomial():
     assert binomial(n, n + 1).func == binomial  # e.g. (-1, 0) == 1
     assert binomial(kp, kp + 1) == 0
     assert binomial(n, u).func == binomial
-    assert binomial(kp, u) == 0
+    assert binomial(kp, u).func == binomial
     assert binomial(n, p).func == binomial
     assert binomial(n, k).func == binomial
     assert binomial(n, n + p).func == binomial
-    assert binomial(kp, kp + p) == 0
+    assert binomial(kp, kp + p).func == binomial
```

Comment:
Since `p` is a positive real therefore `(kp+p).is_integer` is None. Same for above case.